unity-environment/Assets/ML-Agents/Editor/BrainEditor.cs

```diff
@@ -80,13 +75,49 @@ public override void OnInspectorGUI()
                     "Corresponds to whether state" +
                     " vector contains a single integer (Discrete) " +
                     "or a series of real-valued floats (Continuous)."));
-
-                SerializedProperty bpVectorActionSize =
-                    serializedBrain.FindProperty("brainParameters.vectorActionSize");
-                EditorGUILayout.PropertyField(bpVectorActionSize, new GUIContent("Space Size",
-                    "Length of action vector " +
-                    "for brain (In Continuous state space)." +
-                    "Or number of possible values (In Discrete action space)."));
+                if (bpVectorActionType.enumValueIndex == 1)
+                {
+                    //Continuous case :
+                    SerializedProperty bpVectorActionSize =
+                        serializedBrain.FindProperty("brainParameters.vectorActionSize");
+                    bpVectorActionSize.arraySize = 1;
+                    SerializedProperty continuousActionSize =
+                        bpVectorActionSize.GetArrayElementAtIndex(0);
+                    EditorGUILayout.PropertyField(continuousActionSize, new GUIContent(
+                        "Space Size", "Length of continuous action vector."));
+                    
+                }
+                else
+                {
+                    // Discrete case :
+                    SerializedProperty bpVectorActionSize =
+                        serializedBrain.FindProperty("brainParameters.vectorActionSize");
+                    bpVectorActionSize.arraySize = EditorGUILayout.IntField(
+                        "Nb Branches", bpVectorActionSize.arraySize);
```

Comment:
Also another side note, is there a way for us to have a toggle so that our users will still have the our previous display, and if they check the toggle "enable multi discrete action", they will have this new display. 

The reason I bring this up is that having "Nb Branches" under "Space Type" (our new UI) is unintuitive compare to having "Space Size" under "Space Type" (our previous UI). And I saw so many people give up trying or continue to use ml-agents because they have backward compatibility issues (I lurk in some of the ml-agents user group chat and watch them discuss about our product).

Reply:
Unfortunately, the gaol of this PR is to replace entirely the way we do Discrete Control, so we will have back-compatibility issue.